contracts/singletons/SystemDeployerEventEmitterV1.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.30;
+
+import {ISystemDeployerEventEmitterV1} from "../interfaces/decent/singletons/ISystemDeployerEventEmitterV1.sol";
+import {IVersion} from "../interfaces/decent/deployables/IVersion.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+contract SystemDeployerEventEmitterV1 is
+    ISystemDeployerEventEmitterV1,
+    IVersion,
+    ERC165
+{
+    // ======================================================================
+    // ISystemDeployerEventEmitter
+    // ======================================================================
+
+    // --- State-Changing Functions ---
+
+    function emitSystemDeployed(
+        address safeProxyFactory_,
+        bytes32 salt_,
+        bytes calldata initData_
+    ) public virtual override {
+        emit SystemDeployed(msg.sender, safeProxyFactory_, salt_, initData_);
+    }
```

Comment:
One check we could do here to make it harder for someone to manipulate this, is call `deploymentBlock()` on msg.sender (the new Safe), make sure its equal to current block.